### PR TITLE
Register new attribute when user clicks save / edit an attribute

### PR DESCRIPTION
**Pull request by devjoaov**
[linear issue](https://linear.app/bleu-llc/issue/BAL-104/register-new-row-when-user-clicks-save-item-in-modal-form)

### CODE REVIEW

File: packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx [hunk 1]
```diff
@@ -1,54 +1,22 @@
...
-    name: "Pool image",
-    type: "image",
-    desc: "balancer logo",
-    value: "https://s2.coinmarketcap.com/static/img/coins/200x200/5728.png",
-  },
-];
+import {
+  PoolMetadataAttribute,
+  PoolMetadataContext,
+} from "../../../contexts/PoolMetadataContext";
```
Comment (mendesfabio) [inline packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx:19]: minor thing but could you create an issue to add [path aliases](https://nextjs.org/docs/advanced-features/module-path-aliases) to the project?

File: packages/next-app/src/contexts/PoolMetadataContext.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+export const PoolMetadataContext = createContext({} as PoolMetadataContextType);
+
+export function PoolMetadataProvider({ children }: { children: ReactNode }) {
+  const [selectedPool, setSelectedPool] = useState<string>("");
+  // const { data: poolMetadata } = metadataGql.useMetadataPool({
+  //   poolId: selectedPool,
+  // });
+  // cellData = getDataFromIpfs(poolMetadata.CID)
```
Comment (mendesfabio) [inline packages/next-app/src/contexts/PoolMetadataContext.tsx:61]: is it intended to get current metadata if it exists? will be done in another issue?

File: packages/next-app/src/components/PoolMetadataFormModal.tsx [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
-      />
-    </div>
-  );
-}
+type Inputs = {
+  name: string;
+  desc: string;
+  type: string;
+  value: unknown;
+};
```
Comment (mendesfabio) [inline packages/next-app/src/components/PoolMetadataFormModal.tsx:21]: couldn't you import `PoolMetadataAttribute`?

File: packages/next-app/src/contexts/PoolMetadataContext.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import { createContext, ReactNode, useState } from "react";
+
+// import metadataGql from "../lib/poolMetadataGql";
+
+const cellData: {
+  id: string;
+  name: string;
+  type: string;
+  desc: string;
+  value: React.ReactNode;
```
Comment (mendesfabio) [inline packages/next-app/src/contexts/PoolMetadataContext.tsx:13]: couldn't you import `PoolMetadataAttribute`?

File: packages/next-app/src/components/PoolMetadataFormModal.tsx [hunk 5]
```diff
@@ -28,10 +44,34 @@ const inputTypes = [
 
 export function PoolMetadataFormModal({
   children: trigger,
+  context,
+  data,
 }: {
   children: React.ReactNode;
+  context: "add" | "edit";
```
Comment (mendesfabio) [inline packages/next-app/src/components/PoolMetadataFormModal.tsx:51]: minor thing but `context` might be confusing here. maybe `mode` or `type` is best suited to describe it?

File: packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx (hunk 1 above)
Comment (devjoaov) [reply packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx:19]: Yep, thanks 👍

File: packages/next-app/src/components/PoolMetadataFormModal.tsx (hunk 3 above)
Comment (devjoaov) [reply packages/next-app/src/components/PoolMetadataFormModal.tsx:21]: `PoolMetadataAttribute` have an id, but the form input does not have an id. But TBH, it makes sense to add a hidden input field.

File: packages/next-app/src/components/PoolMetadataFormModal.tsx (hunk 5 above)
Comment (devjoaov) [reply packages/next-app/src/components/PoolMetadataFormModal.tsx:51]: agreed. I think `mode` is a good option. thanks sir

File: packages/next-app/src/components/PoolMetadataFormModal.tsx (hunk 3 above)
Comment (mendesfabio) [reply packages/next-app/src/components/PoolMetadataFormModal.tsx:21]: cool! just remember that you can always use the utility-types [Pick](https://www.livelo.com.br/ganhe-pontos-compre-pontue-promo) and [Omit](https://www.livelo.com.br/ganhe-pontos-compre-pontue-promo) to construct new types from another one and avoid code duplication.

Comment (mendesfabio) [reply packages/next-app/src/components/PoolMetadataFormModal.tsx]: cool! just remember that you can always use the utility-types [Pick](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) and [Omit](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) to construct new types from another one and avoid code duplication.

File: packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx (hunk 1 above)
Comment (ribeirojose) [reply packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx:19]: You can use the same convention adopted by Remix here:
```
# tsconfig.json
...
    "paths": {
      "~/*": ["./app/*"]
    },
...
```

**Pull request merged**